### PR TITLE
[2.0.1] Don't use IProperty.Relational().ColumnType when we want only explicitly configured values

### LINKED ISSUE #9344: Migrations after Rename field with Datetime type doesn't work
I just renamed field in many tables with type DateTime
Then I add a migration but the Up() method is empty
but it works well with string fields

### Steps to reproduce
Include a complete code listing (or project/solution) that we can run to reproduce the issue.
Rename a field Datetime 
Add a migration

Partial code listings, or multiple fragments of code, will slow down our response or cause us to push the issue back to you to provide code to reproduce the issue.

```c#
//before modification:
  public DateTime CreatedDateFrm { get; set; }
//after modification:
  public DateTime CreatedFrm { get; set; }
```
In command shell:
`dotnet ef migrations add "data_5" -c Db -v -p ..\GrrDev2.Data\GrrDev2.Data.csproj`

Migration class:

```
 public partial class data_5 : Migration
    {
        protected override void Up(MigrationBuilder migrationBuilder)
        {

        }

        protected override void Down(MigrationBuilder migrationBuilder)
        {

        }
    }
```

### Further technical details
EF Core version: (found in project.json or packages.config)
Microsoft.EntityFrameworkCore (2.0.0)
Database Provider: 
Microsoft.EntityFrameworkCore.SqlServer (2.0.0)
Operating system: 

```
Microsoft Visual Studio Enterprise 2017 Preview
Version 15.3.0 Preview 7.0
VisualStudio.15.Preview/15.3.0-pre.7.0+26730.0
Microsoft .NET Framework
Version 4.7.02542

```
```
.NET Command Line Tools (2.1.0-preview1-007012)

Product Information:
 Version:            2.1.0-preview1-007012
 Commit SHA-1 hash:  b60272d64d

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.16257
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\2.1.0-preview1-007012\

Microsoft .NET Core Shared Framework Host

  Version  : 2.0.0
  Build    : e8b8861ac7faf042c87a5c2f9f2d04c98b69f28d
```

### PR DESCRIPTION
**Pull request by bricelam**
Fixes #9344

### CODE REVIEW

File: test/EFCore.Relational.Tests/Migrations/Internal/MigrationsModelDifferTest.cs [hunk 1]
```diff
@@ -761,6 +764,46 @@ public void Rename_property_and_column()
...
+                        .ValueGeneratedOnAdd();
+
+                    x.HasKey("CrabId");
+                });
+
+            var targetModelBuilder = CreateModelBuilder();
+            targetModelBuilder.Entity<Crab>();
+
+            // NB: Call Validate() so ModelBuilt conventions are applied.
+            targetModelBuilder.GetInfrastructure().Metadata.Validate();
```
Comment (ajcvickers) [inline test/EFCore.Relational.Tests/Migrations/Internal/MigrationsModelDifferTest.cs]: File issue for this?

File: src/EFCore.Relational/Internal/RelationalPropertyExtensions.cs [hunk 2]
```diff
@@ -20,5 +20,12 @@ public static class RelationalPropertyExtensions
         /// </summary>
         public static string FormatColumns([NotNull] this IEnumerable<IProperty> properties)
             => "{" + string.Join(", ", properties.Select(p => "'" + p.Relational().ColumnName + "'")) + "}";
+
+        /// <summary>
+        ///     This API supports the Entity Framework Core infrastructure and is not intended to be used
+        ///     directly from your code. This API may change or be removed in future releases.
+        /// </summary>
+        public static string GetConfiguredColumnType([NotNull] this IProperty property)
+            => (string)property[RelationalAnnotationNames.ColumnType];
```
Comment (ajcvickers) [inline src/EFCore.Relational/Internal/RelationalPropertyExtensions.cs]: Quirk

File: test/EFCore.Relational.Tests/Migrations/Internal/MigrationsModelDifferTest.cs (hunk 1 above)
Comment (bricelam) [reply test/EFCore.Relational.Tests/Migrations/Internal/MigrationsModelDifferTest.cs]: @AndriySvyryd Did we decide we already had an issue for this?

Comment (AndriySvyryd) [reply test/EFCore.Relational.Tests/Migrations/Internal/MigrationsModelDifferTest.cs]: https://github.com/aspnet/EntityFrameworkCore/issues/4016

File: src/EFCore.Relational/Internal/RelationalPropertyExtensions.cs (hunk 2 above)
Comment (bricelam) [reply src/EFCore.Relational/Internal/RelationalPropertyExtensions.cs]: #9610

**Pull request merged**